src/runtime_src/hip/core/event.h

```diff
@@ -124,17 +128,22 @@ class kernel_start : public command
   bool wait() override;
 };
 
+template<class T>
```

Comment:
This class and others and entire file is lacking comments.   What is the responsibility of `copy_buffer` and why is the class templated.  I am not saying this is wrong, I am just trying to understand.